src/actions/settings/currentNetworkActions.js

```diff
@@ -2,13 +2,45 @@ import { createActions } from 'spunky';
 import { getStorage, setStorage } from '../../lib/storage';
 
 export const ID = 'currentNetwork';
+export const NETWORKS_ID = 'networks';
 
+// Setters
 export const setCurrentNetwork = createActions(ID, (currentNetwork) => async () => {
   await setStorage(ID, currentNetwork);
   return currentNetwork;
 });
 
+export const setNetworks = createActions(NETWORKS_ID, (networks) => async () => {
+  await setStorage(NETWORKS_ID, networks);
+  return networks;
+});
+
+export const addNetwork = createActions(NETWORKS_ID, (network) => async () => {
+  const networks = await getStorage(NETWORKS_ID);
+  const newNetworks = [...networks, network];
+  await setStorage(NETWORKS_ID, newNetworks);
+  return newNetworks;
+});
+
+
+export const clearNetworks = createActions(NETWORKS_ID, () => async () => {
+  await setStorage(NETWORKS_ID, []);
+  return [];
+});
+
+// Getters
 export default createActions(ID, () => async () => {
   const currentNetwork = await getStorage(ID);
-  return typeof currentNetwork === 'string' ? currentNetwork : 'TestNet';
+  if (typeof currentNetwork !== 'string') {
+    return 'TestNet';
+  }
+  return currentNetwork;
+});
+
+export const getAllNetworks = createActions(NETWORKS_ID, () => async () => {
```

Comment:
I would consider making this the default export in the new file this gets moved to, but I don't feel strongly either way.